rfcs/0000-central-station.md

```diff
@@ -0,0 +1,507 @@
+(e.g. the name of the person driving the bus) from TransitMaster, for Skate to consume. I believe
+this data is passed along in one of the TripUpdates files (?), and stripped out later in the
+process by concentrate.
+
+### With CentralStation
+
+Busloc will be pared down to something more Trike-esque. Since on-premise Windows servers are so
+hard to deal with, the smaller Busloc is the better. It will continue to handle the TransitMaster
+integration, but it will put this data into Kinesis for CentralStation to consume.
+
+The Samsara feeds will be removed, as those will go directly to CentralStation.
+
+## Trainloc / CommuterRailBoarding
+
+### Current description
+
+TrainLoc and CommuterRailBoarding are two applications that once were separate but were
+consolidated into a single umbrella application, using the name Commuter Rail Boarding, and they
+manage the realtime data of commuter rail trains from Keolis.
+
+Keolis offers two firebase feeds, consumed by CRB:
+
+- The "boarding status" feed, which we get for trains at a few stations, with info like "now
+  boarding on track 9". We put these in TripUpdates_enhanced.json, in our extension
+  `boarding_status` field. Concentrate generally merges these with predictions for the trains, but
+  we often end up with StopTimeUpdates with null arrivals and departures.
+- The vehicle positions feed. We get updates every ~15 seconds with the position of every commuter
+  rail train. If the AVL equipment is working and the train is logged in, it includes the GTFS
+  trip short name. Otherwise we get lat/long and a vehicle ID, but not the trip.
+
+The VehiclePositions are consumed by Swiftly to produce our predictions. The trains without a trip
+short name get auto-assigned a GTFS trip by Swiftly, and Keolis customer service people log into
+the Swiftly dashboard to manually re-tag trains that have been given an incorrect assignment.
+
+### With CentralStation
+
+The vehicle positions feed will go straight to CentralStation. At least initially, the boarding
+status feed will continue to be handled by CRB and continue to go to concentrate as TripUpdates.
+In other words CRB will become CRB again, and drop TrainLoc entirely.
+
+## Glides
+
+### Current description
+
+Right now Glides is just a consumer of the realtime data, directly from RTR.
+
+### With CentralStation
+
+Glides will either get all its data from CentralStation, or get the public data from our public
+feeds, and supplement it with private data from CentralStation.
+
+It will use CentralStation's API for train tagging and service changes.
+
+## Skate
+
+### Current description
+
+Skate consumes bus realtime data, straight from Busloc. This data is GTFS-rt-esque, enriched with
+operator information and block and run data.
+
+### With CentralStation
+
+Skate will get all its vehicle position data from public feeds. It will supplement it with private
```

Comment:
**overall thoughts about being a single source of truth:**

Yes! I love this proposal. I have been frustrated by the lack of a single source of truth for internal data for a while. I think Central Station should lean into that, with the goal of Glides/Skate not needing to ask any other source for realtime data. So publishing an internal extended feed, instead of just a supplement. That way, they don't have to deal with merging and matching data, or with data from different sources being slightly out of sync. My dream is for Central Station to publish a "public_gtfs_rt", a "glides_gtfs_rt_with_light_rail_and_some_internal_data.json", and a "skate_gtfs_rt_with_bus_and_other_internal_data.json".

I think that's best if Central Station plans to envelop all of concentrate and alerts, and become the final aggregator/publisher of data.

Reply:
> (Side note: a single source of truth for schedule data (combining GTFS+HASTUS) might be even more important than realtime data, but I think it's okay if schedule data stays separate from realtime data, and a schedule SSOT would be a separate project.)

For what it's worth, my hope is that the ODS project might provide the first step towards accomplishing this, though it won't be simple. The code to squish together the GTFS and HASTUS schedules is definitely one of the more complicated parts of Skate, not to mention the startup overhead it introduces.

Reply:
One of the meta-level thoughts I had on this RFC is that it says surprisingly little about how "static" data fits into this picture, even as we are looking to introduce features that involve more extensive integration of "static" with real-time data. Contra @skyqrose I might even propose that a single-source-of-truth for _all_ data, whether it is deemed "real-time" or not by the GTFS standard, would be worthwhile to consider. But I think making this a manageable amount of work would also require a fundamentally different approach, and I have enough thoughts here that it could honestly be its own competing RFC.